src/main/java/com/fabriccommunity/spookytime/mixin/BlockMixin.java

```diff
@@ -0,0 +1,21 @@
+package com.fabriccommunity.spookytime.mixin;
+
+import com.fabriccommunity.spookytime.registry.SpookyBlocks;
+import net.minecraft.block.Block;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+/**
+ * Makes deceased soil count as "natural" for tree spawning
+ * @author Indigo Amann
+ */
+@Mixin(Block.class)
+public class BlockMixin {
+
+    @Inject(method = "isNaturalDirt", at = @At("HEAD"), cancellable = true)
+    private static void yesItsNatural(Block block, CallbackInfoReturnable<Boolean> cir) {
+        if (block == SpookyBlocks.DECEASED_GRASS_BLOCK || block == SpookyBlocks.DECEASED_DIRT) cir.setReturnValue(true);
```

Comment:
The vanilla method doesn't return true for Grass Blocks, so why should this return true for Deceased Grass Blocks?

Reply:
@Shnupbups that way dark oak trees can grow on it. mabe i should mixin directly to dark oak, but this fixes it for all trees, if you want to add them

Reply:
My personal take on this issue is that custom features should be made that allow spawning on our custom grass rather than using a mixin in this case. The only reason PlantBlockMixin exists is so tall grass and other plants were able to be placed manually on our grass.

Reply:
ok i can just change it to dark oak